pkg/codegen/go/gen.go

```diff
@@ -2594,35 +2591,80 @@ func generatePackageContextMap(tool string, pkg *schema.Package, goInfo GoPackag
 	// compute set of names generated by a type
 	// handling any potential collisions via remapping along the way
 	scanType := func(t schema.Type) {
+		getNames := func(name, suffix string) []string {
+			names := []string{}
+			names = append(names, name+suffix)
+			names = append(names, name+suffix+"Input")
+			names = append(names, name+suffix+"Output")
+			return names
```

Comment:
Not super important--especially since this is just code getting moved around--but returning a slice literal is a bit shorter and can avoid unnecessary allocations from `append`:

```suggestion
			return []string{name+suffix, name+suffix+"Input", name+suffix+"Output"}
```